### ISSUE TITLE
Task output variables cannot share name with input variables

### ISSUE THREAD

**Issue by brendanreardon** [openwdl/wdl#506]
Hi team,

Is it known that task output variable names cannot also be the name of an input variable? I was unable to find note of this either in the WDL 1.0 spec or issues in this repository. If this occurs, womtool validate will return the error, 
> Failed to process task definition '{task name}' (reason 1 of 1): Duplicate generated values found in: {union of input and output variables, sorted alphabetically}

I discovered this when attempting to set a task output variable as `bam` in the same task that takes a `bam` as an input.

**Comment by mlin** [comment]
Miniwdl also disallows this; I do find it annoying on occasion, but I think the pickle is that an output declaration can use other output declarations, so if an output name was allowed to collide with some other input/intermediate declaration, then we'd need to pick one. (Bearing in mind that WDL also allows forward references!)

**Comment by jdidion** [comment]
Yes, I believe it was ambiguous in the 1.0 spec but it is clearly stated in the 1.1 spec.

**Comment by brendanreardon** [comment]
Ah okay, thank you both for sharing. I appreciate it. Closing this inquiry with this comment then.
